app/controllers/miq_ae_class_controller.rb

```diff
@@ -1815,6 +1816,31 @@ def namespace
     render :json => find_record_with_rbac(MiqAeNamespace, params[:id]).attributes.slice('name', 'description', 'enabled')
   end
 
+  def ae_domains
+    domains = MiqAeDomain.where("ancestry is null and enabled = ?", true).order("name").select("id, name")
+    render :json => {:domains => domains}
+  end
+
+  def ae_methods
+    methods = MiqAeMethod
+              .name_path_search(params[:search])
+              .domain_search(params[:domain_id])
+              .selected_methods(params[:ids])
+              .select("id, relative_path, name")
+              .order('name')
+    render :json => {:methods => methods}
+  end
+
+  def ae_method_operations
+    @edit[:new][:embedded_methods] = MiqAeMethod.selected_methods(params[:ids]).map { |method| "/#{method.relative_path}" }
```

Comment:
```suggestion
    @edit[:new][:embedded_methods] = MiqAeMethod.selected_methods(params[:ids]&.split(',')).pluck(:relative_path).map { |path| "/#{path}" }
```

Reply:
I am not sure we want `selected_methods` - think actually just using `where` is the best route.

- Do we want to bring back every method if the id list is blank?
- Doesn't rails do the split on `","` for us? Or is this a react thing

```suggestion
    @edit[:new][:embedded_methods] = MiqAeMethod.where(:id => params[:ids]&.split(',')).pluck(:relative_path).map { |path| "/#{path}" }
```